### ISSUE TITLE
Output of URLs with valid, but non-stylistic syntax should be rebased and optimised

### ISSUE THREAD

**Issue by kasper** [jakubpawlowicz/clean-css#357]
The following URLs are valid CSS (according to W3C CSS), but *clearly non-stylistic*. When processed through clean-css, the generated output could be optimised regarding quotes, whitespaces and grouping.

**Input**
```css
/* Assets with non-stylistic syntax */

.asset {

    background-image:url('images/image.png');

}

.asset-2 {

    background-image:    url('images/image.png');

}

.asset-3 {

    background-image:
    url('images/image.png');

}

.asset-4 {

    background-image: url(    'images/image.png');

}

.asset-5 {

    background-image: url(
'images/image.png');

}

.asset-6 {

    background-image: url('    images/image.png');

}

.asset-7 {

    background-image: url('images/image.png    ');

}

.asset-8 {

    background-image: url('images/image.png'    );

}

.asset-9 {

    background-image: url('images/image.png'
);

}

.asset-10 {

    background-image: url('images/image.png')    ;

}

.asset-11 {

    background-image: url('images/image.png')
;

}

.asset-12 {

    background-image: URL('images/image.png');

}
```

**Output**
```css
.asset,.asset-2,.asset-3{background-image:url(images/image.png)}.asset-4{background-image:url('images/image.png')}.asset-5{background-image:url('images/image.png')}.asset-6{background-image:url('    images/image.png')}.asset-7{background-image:url('images/image.png    ')}.asset-8{background-image:url('images/image.png')}.asset-9{background-image:url('images/image.png')}.asset-10,.asset-11{background-image:url(images/image.png)}.asset-12{background-image:URL('images/image.png')}
```

**Comment by kasper** [comment]
@jakubpawlowicz Hi! This seems to be mostly fixed in the latest builds. However, the following could be still optimised.

**Input**
```css
.asset-12 {

    background-image: URL('images/image.png');

}
```

**Output**
```css
.asset-12{background-image:URL('/assets/images/image.png')}
```

Notice the uppercase text and redundant single quotes. All of the other cases seem to be fixed.

**Comment by jakubpawlowicz** [comment]
:+1: thanks for rechecking all the cases! It's fixed in master now.

**Comment by kasper** [comment]
@jakubpawlowicz Thanks! (As a side note, they still could be more aggressively grouped, because they have same values, but that’s an other issue.)

**Comment by jakubpawlowicz** [comment]
We should do the normalization at an earlier stage to group them properly. See #432.

**Comment by kasper** [comment]
@jakubpawlowicz Sounds good!

**Comment by jakubpawlowicz** [comment]
#432 is fixed on master now and due in 4.0 release.
